mmdet/datasets/pipelines/cutmix.py

```diff
@@ -0,0 +1,161 @@
+            lam (float): Cutmix lambda value
+            margin (float): Percentage of bbox dimension to enforce as margin
+                (reduce amount of box outside image). Default to 0.
+            count (int, optional): Number of bbox to generate. Default to None
+        """
+        ratio = np.sqrt(1 - lam)
+        img_h, img_w = img_shape[-2:]
+        cut_h, cut_w = int(img_h * ratio), int(img_w * ratio)
+        margin_y, margin_x = int(margin * cut_h), int(margin * cut_w)
+        cy = np.random.randint(0 + margin_y, img_h - margin_y, size=count)
+        cx = np.random.randint(0 + margin_x, img_w - margin_x, size=count)
+        yl = np.clip(cy - cut_h // 2, 0, img_h)
+        yh = np.clip(cy + cut_h // 2, 0, img_h)
+        xl = np.clip(cx - cut_w // 2, 0, img_w)
+        xh = np.clip(cx + cut_w // 2, 0, img_w)
+        return yl, yh, xl, xh
+
+    def cutmix_bbox_and_lam(self, img_shape, lam, count=None):
+        """Generate bbox and apply lambda correction.
+
+        Args:
+            img_shape (tuple): Image shape as tuple
+            lam (float): Cutmix lambda value
+            count (int, optional): Number of bbox to generate. Default to None
+        """
+        if self.cutmix_minmax is not None:
+            yl, yu, xl, xu = self.rand_bbox_minmax(img_shape, count=count)
+        else:
+            yl, yu, xl, xu = self.rand_bbox(img_shape, lam, count=count)
+        if self.correct_lam or self.cutmix_minmax is not None:
+            bbox_area = (yu - yl) * (xu - xl)
+            lam = 1. - bbox_area / float(img_shape[-2] * img_shape[-1])
+        return (yl, yu, xl, xu), lam
+
+    def one_hot_encoding(self, gt, num_classes):
+        """Change gt_label to one_hot encoding.
+
+        If the shape has 2 or more
+        dimensions, return it without encoding.
+        Args:
+            gt (Tensor): The gt label with shape (N,) or shape (N, */).
+            num_classes (int): The number of classes.
+        Return:
+            Tensor: One hot gt label.
+        """
+        if gt.ndim == 1:
+            # multi-class classification
+            return F.one_hot(gt, num_classes=num_classes)
+        else:
+            # binary classification
+            # example. [[0], [1], [1]]
+            # multi-label classification
+            # example. [[0, 1, 1], [1, 0, 0], [1, 1, 1]]
+            return gt
+
+    def cutmix(self, img, gt_label):
+        one_hot_gt_label = self.one_hot_encoding(gt_label, self.num_classes)
+        lam = np.random.beta(self.alpha, self.alpha)
+        batch_size = img.size(0)
+        index = torch.randperm(batch_size)
+
+        (bby1, bby2, bbx1,
+         bbx2), lam = self.cutmix_bbox_and_lam(img.shape, lam)
+        img[:, :, bby1:bby2, bbx1:bbx2] = \
+            img[index, :, bby1:bby2, bbx1:bbx2]
+        mixed_gt_label = lam * one_hot_gt_label + (
+            1 - lam) * one_hot_gt_label[index, :]
+        return img, mixed_gt_label
+
+    def __call__(self, img, gt_label):
```

Comment:
Maybe you can refer to   https://github.com/open-mmlab/mmdetection/blob/6b87ac22b8d9dea8cc28b9ce84909e6c311e6268/mmdet/datasets/pipelines/transforms.py#L2295
The input and output for data augmentation in `mmdet/pipelines/transforms.py` should be a dict named `results`, and cutmix is a kind of data augmentation for classification, there should some difference for object detection, considering gt_bboxes.